### PR TITLE
Edit an entry

### PR DESCRIPTION
**Pull request by alexjraymond**
CleanShot 2023-01-02 at 14 34 51

--stuck on the refresh part to update the dom/list, but other than that i think it's there

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -14,26 +14,46 @@ var $entryList = document.querySelector('ul');
...
+    var editingIndex = data.entries.entryId;
+    var editedEntry = {
+      entryId: editingIndex,
+      note: $form.elements.notes.value,
+      link: $form.elements.photourl.value,
+      title: $form.elements.title.value
+    };
+
+    data.entries.splice(editingIndex);
+    data.entries.unshift(editedEntry);
```
Comment (codingcodymiller) [inline js/main.js]: You don't want to add edited entries to the beginning of the list, you want to replace them at the index they were at before editing.

File: js/main.js [hunk 2]
```diff
@@ -14,26 +14,46 @@ var $entryList = document.querySelector('ul');
...
+    toggleNoEntries();
+    viewSwap('entries');
+    var editedLiReturn = renderEntry(editedEntry);
+    $entryList.appendChild(editedLiReturn);
```
Comment (codingcodymiller) [inline js/main.js]: Same here, you shouldn't be adding it to the end of the list, you should be replacing the original DOM tree at its original position before editing.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by alexjraymond** [comment]
CleanShot 2023-01-03 at 15 19 38

File: js/main.js [hunk 3]
```diff
@@ -14,26 +14,49 @@ var $entryList = document.querySelector('ul');
...
+    for (var i = 0; i < $allTheLi.length; i++) {
+      var individualLi = Number($allTheLi[i].getAttribute('data-entry-id'));
+      if (data.editing.entryId === individualLi) {
+        $allTheLi[i].replaceWith(renderEntry(editedEntry));
+      }
+    }
+    toggleNoEntries();
+    viewSwap('entries');
+    data.editing = null;
+  }
```
Comment (codingcodymiller) [inline js/main.js]: It doesn't look like you are updating the edited entry's data in `data.entries`. Does your edit persist after you refresh the page?

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by alexjraymond** [comment]
CleanShot 2023-01-09 at 14 49 46@2x

we did it

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**